src/main/java/io/vertx/core/http/impl/Http1xPool.java

```diff
@@ -81,39 +83,40 @@ public HttpClientStream createStream(ClientConnection conn) {
     return conn;
   }
 
-  // Called when the request has ended
   public void recycle(ClientConnection conn) {
-    if (pipelining) {
-      doRecycle(conn);
-    }
-  }
-
-  // Called when the response has ended
-  public void responseEnded(ClientConnection conn, boolean close) {
-    if ((pipelining || keepAlive) && !close && conn.getCurrentRequest() == null) {
-      doRecycle(conn);
-    } else {
-      // Close it now
-      conn.close();
-    }
-  }
-
-  private void doRecycle(ClientConnection conn) {
     synchronized (queue) {
       Waiter waiter = queue.getNextWaiter();
       if (waiter != null) {
-        Context context = waiter.context;
```

Comment:
when we recycle, we should really use the Context of the connection + now the runOnContext() is done before calling recycle (because it should not be done in some cases) rather than in recycle.